cpp/src/arrow/filesystem/util_internal.h

```diff
@@ -99,6 +99,16 @@ ARROW_EXPORT
 Result<FileInfoVector> GlobFiles(const std::shared_ptr<FileSystem>& filesystem,
                                  const std::string& glob);
 
+/// \brief Decode the Input SSE key,calculate the MD5
+/// \param base64_encoded_key is the input base64 encoded sse key
+/// \param md5_result, output resut
+/// \param expect_input_key_size, default 32
+/// \return true if the decode and calculate MD5 success, otherwise return false
+ARROW_EXPORT
+bool CalculateSSECKeyMD5(const std::string& base64_encoded_key, std::string& md5_result,
```

Comment:
Would you mind change to `Result<std::string>`?

Reply:
ok, I would try to refine the return value for the function.